analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/Binary.cs

```diff
@@ -37,11 +37,12 @@ protected override ProgramState LearnBranchingConstraint(ProgramState state, IBi
             state = LearnBranchingEqualityConstraint<ObjectConstraint>(state, operation, falseBranch) ?? state;
             state = LearnBranchingEqualityConstraint<BoolConstraint>(state, operation, falseBranch) ?? state;
             state = LearnBranchingEqualityConstraint<NumberConstraint>(state, operation, falseBranch) ?? state;
+            state = LearnBranchingNumberConstraint(state, operation, falseBranch) ?? state;
```

Comment:
These two statements do fundamentally the same thing. The second one makes the first one redundant.
We could remove the first statement. But I think merging `LearnBranchingEqualityConstraint` and `LearnBranchingNumberConstraint` into one method is better.

Reply:
They are similar, but they have structural differences. It originally used the same implementation but then deviated too much. Because I can't fit the opposite and flip into the generic method without doing some extra twists there

Reply:
I can remove the first statement, and all tests stay green. At the very least, we should remove the first statement to save performance and avoid confusion.